Core/Core/Models/Assignment.swift

```diff
@@ -114,12 +137,13 @@ extension Assignment {
         hideRubricPoints = item.rubric_settings?.hide_points == true
 
         if updateSubmission {
-            if let submissionItem = item.submission?.values.first {
-                let sub = Submission.save(submissionItem, in: client)
-                submission = sub
-            } else if let submission = submission {
-                client.delete(submission)
-                self.submission = nil
+            if let values = item.submission?.values, values.count > 0 {
+                self.submissions = Set<Submission>()
+                let newSubs = values.map { Submission.save($0, in: client) }
+                self.submissions = Set<Submission>(newSubs)
+            } else if let submissions = submissions {
+                client.delete(Array(submissions))
```

Comment:
Is this needed with `deletionRule="Cascade"`?

Reply:
yeah, there is a test for that and it was not passing w/ out this line.  I do have it set to deletionRule = Cascade